doc/deploying_gcs_on_bare_metal_machines.md

```diff
@@ -0,0 +1,196 @@
+
+    ## Hosts that will run etcd for the Kubernetes cluster
+    [etcd]
+    kube1
+    kube2
+    kube3
+
+    ## Hosts that will be kubernetes master nodes
+    [kube-master]
+    kube1
+
+    ## Hosts that will be kuberenetes nodes
+    [kube-node
+    kube1
+    kube2
+    kube3
+
+    ## The full kubernetes cluster
+    [k8s-cluster:children]
+    kube-master
+    kube-node
+
+    ## Hosts that will be used for GCS.
+    ## Systems grouped here need to define 'gcs_disks' as hostvars, which are the disks that will be used by GCS to 
+    provision storage.
+    [gcs-node]
+    kube1
+    kube2
+    kube3
+    ```
+
+
+6. To deploy kubernetes we need to excecute deploy-k8s.yml file using ansible and provide inventory file created in Step 5.<br/>
+   The deploy-k8s.yml file is present under deploy directory in gcs repository. <br />
+   `$ ansible-playbook -i <path to inventory file>/<name of inventory file>  <path to`
+     `deploy-k8s.yml file>/deploy-k8s.yml`
+
+7. Check whether all the nodes are in ready state.<br />
+   `$ kubectl get nodes`<br />
+
+    The status of all the nodes should be in READY state.
+
+> Note: In case deploy kubernetes fails then please check the troubleshooting section.
+
+### Deploying GCS on kubernetes cluster:
+
+#### Steps to install GCS:
+
+1. To deploy GCS on kubernetes cluster we will execute the deploy-gcs.yml file using ansible from master node(kube 1). The deploy-gcs.yml file is present under deploy directory.<br />
+   `$ ansible-playbook -i <path to inventory file>/<name of inventory file>  <path to deploy-gcs.yml file>/deploy-gcs.yml`.
+
+### Testing the deployment
+
+To test whether the deployment was successfull or not, you can run the following command on the master node.<br />
+`$ kubectl get pods -n gcs`<br />
+
+All the pods should be in `Running` state.
+
+### Steps to create volume from kubernetes.
+
+All commands are to be executed on the master node.<br />
+1. Create a yaml file of the format presented below.<br />
+   ```
+   ---
+   kind: PersistentVolumeClaim
+   apiVersion: v1
+   metadata:
+      name: gcs-pvc
+   spec:
+      storageClassName: glusterfs-csi
+      accessModes:
+         - ReadWriteMany
+      resources:
+        requests:
+           storage: 2Gi
+   ```
+   `gcs-pvc` is name of the pvc that is to be created, and the size of volume is `2Gi`.
+   Name of file : pvc.yaml
+
+2. To create PVC, run the following command.<br />
+   `$ kubectl create -f pvc.yaml`
+
+3. To check whether the PVC is created or not run the following command.
+   `$ kubectl get pvc`
+
+   If the PVC is in BOUND state then the volume is created and can be used now.
+
+#### Creating parallel PVCs:
+
+To create parallel PVCs, run the "pvc create" command in loop without waiting for the previous PVC to be in BOUND state. You will have to change the name of pvc in the pvc.yaml file to create a new pvc.<br />
+
+To create parallel PVC using python:<br />
+```
```

Comment:
Alternatively you can use following script for better readability.

```
import os
from string import Template

claimdata = """---
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: gpv-$pv_number
spec:
  storageClassName: glusterfs-csi
  accessModes:
    - ReadWriteMany
  resources:
    requests:
      storage: 1Gi

"""

start = 1
end = 500
claim_t = Template(claimdata)
claim_f = open("pvc.yaml", "w")
for pv in range(start, end+1):
    claim_f.write(claim_t.substitute({"pv_number": pv}))

claim_f.flush()
claim_f.close()

os.system('kubectl create -f pvc.yaml')
```

Reply:
this creates the multiple PVC of size 1Gib, does this helps the end User?
@aravindavk can we remove this section itself, the end user may use different scripting mechanism based on the knowledge and his requirement,